### PR TITLE
feat: (levm) Implement calldata and returndata opcodes

### LINKED ISSUE #517: Opcode: CALLDATALOAD

### LINKED ISSUE #518: Opcode: CALLDATASIZE

### LINKED ISSUE #519: Opcode: CALLDATACOPY

### LINKED ISSUE #525: Opcode: RETURNDATASIZE

### LINKED ISSUE #526: Opcode: RETURNDATACOPY

### PR DESCRIPTION
**Pull request by iinaki**
**Motivation**
Implemented `CALLDATALOAD`, `CALLDATASIZE`, `CALLDATACOPY`, `RETURNDATASIZE`, `RETURNDATACOPY`

**Description**

Closes #517 
Closes #518 
Closes #519 
Closes #525 
Closes #526

### CODE REVIEW

**Review by lazcanoluca — APPROVED** [review]
Just for the record, we discussed about changing the `Memory` type to plain `Bytes` in `calldata` and `return_data`, which was done.

Approving 🚀

File: crates/levm/src/memory.rs [hunk 1]
```diff
@@ -10,6 +10,10 @@ impl Memory {
         Self { data: Vec::new() }
     }
 
+    pub fn new_from_vec(data: Vec<u8>) -> Self {
+        Self { data }
+    }
```
Comment (ilitteri) [inline crates/levm/src/memory.rs:15]: Can't we implement `From<Vec<u8>> for `Memory``?

**Review by ilitteri — APPROVED** [review]
LGTM

File: crates/levm/src/memory.rs (hunk 1 above)
Comment (iinaki) [reply crates/levm/src/memory.rs]: We can, done!!

**Pull request merged**